apps/angular/signal-input/src/app/user.component.ts

```diff
@@ -27,15 +27,21 @@ const ageToCategory = (age: number): Category => {
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+  // @Input({ required: true }) name!: string;
+  // @Input() lastName?: string;
+  // @Input() age?: string;
+
+  name = input.required<string>();
+  lastName = input<string>('')
```

Comment:
use generic only if the value you give is null or undefined, otherwise the type is infered by TS

Reply:
So, is this correct approach now if I changed based on your suggestions..

`name = input.required<string>('');
  lastName = input<string>()
  age = input<string>()

  fullName = computed(()=> `${this.name()} ${this.lastName()}` );
  category =  computed(() => ageToCategory(Number(this.age())))`